### PR TITLE
add git cli workshop for JS1 week 1

### PR DESCRIPTION
**Pull request by Dedekind561**
Think of this as a guided tour through the CLI operations required to produce a git commit. It will focus heavily on checking the state of the working tree and staging area to really bring these concepts to the fore.

### CODE REVIEW

File: git-cli/committing.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Committing changes
+
+## Learning objectives
+
+- [ ] commit changes in the staging area
+
+
+First off, double check the state of your **Working Tree** and your **Staging Area**
```
Comment (SallyMcGrath) [inline git-cli/committing.md:8]:
```suggestion
First, double check the state of your **Working Tree** and your **Staging Area**
```

File: git-cli/committing.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+Read the docs above to create a commit using the `git commit` command.
+
+
+Now you need to check you successfully created a commit. Which command can you use to check this?
+
+
+## Next steps 👣
+
+Once you've done your first commit for `exercises/ex-1.js` you can then repeat these steps to solve the other exercises.
```
Comment (SallyMcGrath) [inline git-cli/committing.md:21]:
```suggestion
Once you've done your first commit for `exercises/ex-1.js`, repeat these steps to solve the other exercises.
```

File: git-cli/recap.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+It is the directory with the files and folders that we can view and edit.
+When we're making changes to our code in VSCode, we're making changes in the working tree.
+
+To understand what is happening in this workshop, you will need to continually check the **status** of the **Working Tree** and the **Staging Area**.
+
+So let's start with a check...
+
+## 👣 Steps
+
+1. Run the command `git status --help` and check the name and description of this command (You can press `q` to quit)
```
Comment (SallyMcGrath) [inline git-cli/recap.md:24]:
```suggestion
1. Run the command `git status --help` and check the name and description of this command (Press `q` to quit)
```

File: git-cli/staging.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+
+## Updating the working tree
+
+Let's make a change to the current working directory and then stage that change.
+
+
+### Steps 👣
+
+1. Open up this project in VSCode (if you've not done so already).
```
Comment (SallyMcGrath) [inline git-cli/staging.md:17]:
```suggestion
1. Open this project in VSCode.
```

File: git-cli/staging.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+2. Go to `examples/ex-1.js` and try the first exercise in there
```
Comment (SallyMcGrath) [inline git-cli/staging.md:18]:
```suggestion
2. Go to `examples/ex-1.js` and try the first exercis.
```

File: git-cli/staging.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+You will have made a change to the **working tree**.
+
+3. Once you've updated `examples/ex-1.js` go to the Version Control tab in VSCode and double-check you can see your change to the Working Tree.
```
Comment (SallyMcGrath) [inline git-cli/staging.md:22]:
```suggestion
3. Now go to the Version Control tab in VSCode and look for your change to the Working Tree.
```

File: git-cli/staging.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+We do this by promoting these changes files to the **staging area**.
+
+🔑 **Staging area** is the space to prepare (stage) the changes that will be in the next commit.
+
+
+### Steps 👣
+
+1. Use `git status` command to check the state of your **Working Tree** and your **Staging Area**. What does `git status` tell you?
+2. Now **stage** this change using the `git add` command to stage your change to `exercises/ex-1.js`
+3. Once again, after you've done this - check the status of the **Working Tree** and the **Staging Area**.
```
Comment (SallyMcGrath) [inline git-cli/staging.md:38]:
```suggestion
3. Finally, check the status of the **Working Tree** and the **Staging Area**.
```

**Comment by Dedekind561** [comment]
I'm going to go ahead and merge

**Pull request merged**